doc/http-grpc-status-mapping.md

```diff
@@ -0,0 +1,30 @@
+# HTTP to gRPC Status Code Mapping
+
+Since intermediaries are a common part of HTTP infrastructure some responses to
+gRPC requests may be received that do not include the grpc-status header. In
+some cases mapping error codes from an intermediary allows the gRPC client to
+behave more appropriately to the error situation without overloading the
+semantics of either error code.
+
+This table is to be used _only_ for clients that received a response that did
+not include grpc-status. If grpc-status was provided, it _must_ be used. Servers
+_must not_ use this table to determine an HTTP status code to use; the mappings
+are neither symmetric nor 1-to-1.
+
+| HTTP Status Code           | gRPC Status Code   |
+|----------------------------|--------------------|
+| 400 Bad Request            | INTERNAL           |
+| 401 Unauthorized           | UNAUTHENTICATED    |
+| 403 Forbidden              | PERMISSION\_DENIED |
+| 404 Not Found              | UNIMPLEMENTED      |
+| 429 Too Many Requests      | UNAVAILABLE        |
```

Comment:
`UNAVAILABLE` is "most likely a transient condition, which can be corrected by retrying with a backoff." Whereas `RESOURCE_EXHAUSTED` commonly requires service-specific handling to resolve, like when out of disk quota (or even rate limiting). So while, yes, `RESOURCE_EXHAUSTED` could maybe also work, client handling here should be identical to `UNAVAILABLE`. It also still follows the `UNAVAILABLE` semantics, especially when considering the litmus test provided in `FAILED_PRECONDITION`.

For things like this, I'm happy these mappings aren't set in stone and we can change them. I could completely believe it was a mistake.